modules/p2/ding_list/ding_list.module

```diff
@@ -662,14 +662,20 @@ function ding_list_ding_list_add_to_list_data_alter(&$data, $context) {
         );
       }
       elseif (!user_is_logged_in()) {
+        $login_path = '#';
+        if (module_exists('ding_adgangsplatformen')) {
+          $login_path = '/' . DING_ADGANGSPLATFORMEN_LOGIN_URL;
```

Comment:
I recognize the need to provide a new login url when the user is not authenticated but this seems somewhat crude. 

Could this be implemented more elegantly e.g. by adding support for a provider callback which returns a user login url but defaults to `/user` if nothing is provided?

Reply:
Well, yes/no... here it defaults to "#" where the only other place this is need is in the theme where it defaults to "user", but thats not the link but the path to match to fix icons etc.

I'm not sure how to do this and if it makes sens to add the extra layer to fix this one place in ding_list.

Reply:
Oh well. Perhaps it is too complex to clean this up.